src/Reference/HasMany.php

```diff
@@ -14,6 +14,24 @@
  */
 class HasMany extends Reference
 {
+    public function getTheirFieldName(): string
+    {
+        if ($this->their_field) {
+            return $this->their_field;
+        }
+
+        // this is pure guess, verify if such field exist, otherwise throw
+        // TODO probably remove completely in the future
+        $ourModel = $this->getOurModel();
+        $theirField = $ourModel->table . '_' . $ourModel->id_field;
+        if (!$this->createTheirModel()->hasField($theirField)) {
+            throw (new Exception('Their model does not contain fallback field'))
+                ->addMoreInfo('their_fallback_field', $theirField);
+        }
+
+        return $theirField;
```

Comment:
Maybe assign it to `$this->their_field`?

Reply:
And rename `$theirField` to `$theirFieldName` for consistency...

Reply:
no - we should not materialize guessed if not done on construction time